src/components/routerView.browser.spec.ts

```diff
@@ -181,10 +185,13 @@ it('Renders custom NotFound component when the initialUrl does not match', () =>
     },
   })
 
-  expect(app.text()).toBe(template)
+  const route = mount(router.route.matched.component)
+
+  expect(app.text()).toBe(NotFound.template)
+  expect(route.text()).toBe(NotFound.template)
```

Comment:
@stackoverfloweth 
This is the failing test to show what I'm talking about with the rejection route. Both `rejection` and `route` get set. In this branch the rejection is set correctly. But the `route` still has the library defined not found component rather than the user defined one.